distros/humble/overrides.nix

```diff
@@ -130,10 +130,33 @@ in with lib; {
     '';
   });
 
-  rviz-ogre-vendor = patchVendorUrl rosSuper.rviz-ogre-vendor {
-    url = "https://github.com/OGRECave/ogre/archive/v1.12.1.zip";
-    sha256 = "1iv6k0dwdzg5nnzw2mcgcl663q4f7p2kj7nhs8afnsikrzxxgsi4";
-  };
+  # This is a newer version than the build system tries to download,
+  # cleaned definition to fix build for macOS on ARM64 since v1.12.9
+  rviz-ogre-vendor = (patchVendorUrl rosSuper.rviz-ogre-vendor {
+    originalUrl = "https://github.com/OGRECave/ogre/archive/v1.12.1.zip";
+    url = "https://github.com/OGRECave/ogre/archive/v1.12.13.zip";
+    hash = "sha256-S5CTQFiBPTs5w0DI9lpRIixqRBXFTyRs63/lk9uh5vI=";
+  }).overrideAttrs ({
+    propagatedBuildInputs ? [], nativeBuildInputs ? [], patches ? [], postPatch ? "", ...
+  }: {
+    propagatedBuildInputs = propagatedBuildInputs ++ [ self.pugixml self.glew ];
+    nativeBuildInputs = nativeBuildInputs ++ lib.optionals self.stdenv.isDarwin [
+      self.darwin.apple_sdk.frameworks.Foundation
+      self.darwin.apple_sdk.frameworks.AppKit
+    ];
+    patches = patches ++ lib.optionals self.stdenv.isDarwin [
+      # Fix build with darwin-aarch64
+      (self.fetchpatch {
+        url = "https://github.com/ros2-gbp/rviz-release/commit/777d3f45f7d6197e3ee3b7e3b9e95f0138ca6266.patch";
```

Comment:
This is not what I meant. AFAICS the commit you're referring to is only in [your repository](https://github.com/etherswangel/rviz-release/tree/patch/darwin-aarch64). If it's needed, you should at least use the URL of your repo. However, it's not needed in this case, because the same fixes are already merged in the upstream repo (e.g. [here](https://github.com/ros2/rviz/pull/944/files) and [here](https://github.com/ros2/rviz/pull/1156/files)). The preferred way is to use those patches.